### PR TITLE
Add camera-streaming-daemon package

### PR DESCRIPTION
**Pull request by Rjasuja**
Enable Camera streaming daemon
Issue: #15

### CODE REVIEW

File: aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Source: camera-streaming-daemon
+Priority: optional
+Maintainer: Ritul Jasuja <ritul.jasuja@intel.com>
+Build-Depends: debhelper (>=9), dh-systemd (>= 1.5), dh-autoreconf, git, python-future, libavahi-client-dev, libavahi-core-dev, libavahi-glib-dev, libqt5gstreamer-1.0-0, libgstreamer-plugins-base1.0-dev, libgstrtspserver-1.0-dev
```
Comment (anselmolsm) [inline aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control]: libqt5gstreamer ?
Please check https://github.com/01org/camera-streaming-daemon/blob/master/configure.ac for actual list of dependencies.

File: aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Standards-Version: 3.9.7
+Homepage: https://github.com/01org/camera-streaming-daemon
+
+Package: aero-camera-streaming-daemon
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
```
Comment (anselmolsm) [inline aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control]: I bet the way it is now the generated packages won't tell apt about its runtime dependencies; which means csd gets installed without the required libraries.

File: aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/rules [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+override_dh_auto_configure:
+	dh_auto_configure -- --with-systemdsystemunitdir=/lib/systemd/system
+
+override_dh_shlibdeps:
+	dh_shlibdeps --dpkg-shlibdeps-params=--ignore-missing-info
+
+override_dh_fixperms:
+	dh_fixperms
+	chmod 644 config/main.conf
+	chmod 755 config/csd
```
Comment (anselmolsm) [inline aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/rules]: Why does this config file need to be executable?

File: .travis.yml [hunk 4]
```diff
@@ -4,7 +4,7 @@ services: docker
 
 before_install:
  - CONTAINER=$(docker run -it -d -v $PWD:/build ubuntu:16.04 /bin/bash)
- - docker exec -i $CONTAINER /bin/bash -c "apt-get update -qq && apt-get install -y -qq gnupg git equivs lintian devscripts build-essential bc python-future dh-systemd dh-autoreconf wget libglfw3-dev cmake pkg-config libusb-1.0-0-dev"
+ - docker exec -i $CONTAINER /bin/bash -c "apt-get update -qq && apt-get install -y -qq gnupg git equivs lintian devscripts build-essential bc python-future dh-systemd dh-autoreconf wget libglfw3-dev cmake pkg-config libusb-1.0-0-dev libavahi-client-dev libavahi-core-dev libavahi-glib-dev gstreamer-1.0 libgstreamer-plugins-base1.0-dev libgstrtspserver-1.0-dev"
```
Comment (anselmolsm) [inline .travis.yml:7]: The list of packages added to this list differs from the one at https://github.com/intel-aero/packages/pull/16/commits/cbe5d860b32a718c7a573a7ba6fadb82a2d6061d#diff-f1ac8795020ceffc2a0af67bf2b3f5d4 (I am obviously considering only the CSD-specific packages)

**Review by anselmolsm — CHANGES_REQUESTED** [review]

File: aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control (hunk 1 above)
Comment (Rjasuja) [reply aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control]: gstreamer-1.0 is just a list of dependency debian packages which includes libqt5gstreamer, gir1.2-gstreamer-1.0, and libqtgstreamer. To check if this package got installed, i have added one of the dependency package in the Build-Depends list.

File: .travis.yml (hunk 4 above)
Comment (Rjasuja) [reply .travis.yml:7]: This list is as per the dependency list in:  https://github.com/01org/camera-streaming-daemon/blob/master/configure.ac

File: aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control (hunk 1 above)
Comment (anselmolsm) [reply aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control]: Why not gstreamer-1.0 itself?

Comment (anselmolsm) [reply aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control]: ping

File: aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control (hunk 2 above)
Comment (anselmolsm) [reply aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control]: ping

File: aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control (hunk 1 above)
Comment (avinash-palleti) [reply aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control]: As there is no package with the name gstreamer-1.0 everytime this check will fail. So one of the actual package that gets installed is added as dependency.

File: aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control (hunk 2 above)
Comment (avinash-palleti) [reply aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control]: I think @Rjasuja for now verified build time dependencies on docker image, we will run the generated debian package on docker image to know the runtime dependencies.

Comment (Rjasuja) [reply aero-camera-streaming-daemon/aero-camera-streaming-daemon_1.0/debian/control]: Runtime Dependencies are taken care by ${shlibs:Depends} variable. Verified it on fresh ubuntu docker image

**Comment by anselmolsm** [comment]
Rebase needed.

**Pull request merged**